var/spack/repos/builtin/packages/sperr/package.py

```diff
@@ -12,23 +12,32 @@ class Sperr(CMakePackage):
 
     # Package info
     homepage = "https://github.com/NCAR/SPERR"
-    url = "https://github.com/NCAR/SPERR/archive/refs/tags/v0.6.2.tar.gz"
+    url = "https://github.com/NCAR/SPERR/archive/refs/tags/v0.7.1.tar.gz"
     git = "https://github.com/NCAR/SPERR.git"
     maintainers("shaomeng", "robertu94")
 
     # Versions
     version("main", branch="main")
+    version("0.7.1", sha256="1c3f46200be365427d1f57f5873f1b0b6dbcd297de4603a47a7fa3f41b273d79")
     version("0.6.2", sha256="d986997e2d79a1f27146ad02c623359976a1e72a1ab0d957e128d430cda3782d")
     version("0.5", sha256="20ad48c0e7599d3e5866e024d0c49648eb817f72ad5459f5468122cf14a97171")
 
-    depends_on("git", type="build")
-    depends_on("zstd", type=("build", "link"), when="+zstd")
-    depends_on("pkgconfig", type=("build"), when="+zstd")
-
+    # Variants
     variant("shared", description="build shared libaries", default=True)
-    variant("zstd", description="use zstd for more compression", default=True)
-    variant("openmp", description="use openmp in 3D inputs", default=True)
+    variant("openmp", description="use OpenMP in 3D inputs", default=True)
     variant("utilities", description="build SPERR CLI utilities", default=True)
+    variant("zstd", description="use ZSTD for more compression", default=True, when="@:0.6.2")
```

Comment:
So I understand, are you saying `zstd` can now only be enabled for versions up to `0.6.2`?

Reply:
That is what I understand from @shaomeng, but he should confirm.

Reply:
Versions up to `0.6.2` can *optionally* take advantage of `zstd` for a little extra compression, which is configurable through CMake. The benefit is however very minimal, so for simplicity, I removed this layer of ZSTD completely in version `0.7.x` and forward, thus there's no `zstd` dependency whatsoever. (There is nothing to be configured w.r.t. `zstd` after version `0.6.2`.)

I of course want the spack recipe to correctly express the underlying logic, and I really appreciate it if you can confirm that the current constraints are doing so (or not). Thanks!

Reply:
You can get an idea of how Spack resolves dependencies using `spack spec sperr` plus whatever options you want. 

For example, `spack spec sperr+zstd` for me results in 

```
Input spec
--------------------------------
 -   sperr+zstd

Concretized
--------------------------------
 -   sperr@0.6.2%gcc@12.1.1~bundle_zstd~ipo+openmp+shared+utilities+zstd build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64_v3
[+]      ^cmake@3.27.4%gcc@12.1.1~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rhel8-x86_64_v3
[+]          ^curl@7.88.1%gcc@12.1.1~gssapi~ldap+libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-rhel8-x86_64_v3
[+]              ^nghttp2@1.52.0%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^ncurses@6.4%gcc@12.1.1~symlinks+termlib abi=none build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^zlib@1.2.13%gcc@12.1.1+optimize+pic+shared build_system=makefile arch=linux-rhel8-x86_64_v3
[+]      ^git@2.41.0%gcc@12.1.1+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^autoconf@2.69%gcc@12.1.1 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-rhel8-x86_64_v3
[+]          ^automake@1.16.5%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^expat@2.5.0%gcc@12.1.1+libbsd build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^libbsd@0.11.7%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]                  ^libmd@1.0.4%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^gettext@0.21.1%gcc@12.1.1+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^bzip2@1.0.8%gcc@12.1.1~debug~pic+shared build_system=generic arch=linux-rhel8-x86_64_v3
[+]              ^libxml2@2.10.3%gcc@12.1.1~python build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^tar@1.34%gcc@12.1.1 build_system=autotools zip=pigz arch=linux-rhel8-x86_64_v3
[+]                  ^pigz@2.7%gcc@12.1.1 build_system=makefile arch=linux-rhel8-x86_64_v3
[+]              ^xz@5.4.1%gcc@12.1.1~pic build_system=autotools libs=shared,static arch=linux-rhel8-x86_64_v3
[+]          ^libiconv@1.17%gcc@12.1.1 build_system=autotools libs=shared,static arch=linux-rhel8-x86_64_v3
[+]          ^libidn2@2.3.4%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^libunistring@1.1%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^libtool@2.4.7%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^m4@1.4.19%gcc@12.1.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rhel8-x86_64_v3
[+]              ^diffutils@3.9%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^libsigsegv@2.14%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^openssh@9.3p1%gcc@12.1.1+gssapi build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^krb5@1.20.1%gcc@12.1.1+shared build_system=autotools arch=linux-rhel8-x86_64_v3
[+]                  ^bison@3.8.2%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]                  ^findutils@4.9.0%gcc@12.1.1 build_system=autotools patches=440b954 arch=linux-rhel8-x86_64_v3
[+]              ^libedit@3.1-20210216%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]              ^libxcrypt@4.4.35%gcc@12.1.1~obsolete_api build_system=autotools patches=4885da3 arch=linux-rhel8-x86_64_v3
[+]          ^openssl@3.1.1%gcc@12.1.1~docs~shared build_system=generic certs=system arch=linux-rhel8-x86_64_v3
[+]          ^pcre2@10.42%gcc@12.1.1~jit+multibyte build_system=autotools arch=linux-rhel8-x86_64_v3
[+]          ^perl@5.38.0%gcc@12.1.1+cpanm+opcode+open+shared+threads build_system=generic arch=linux-rhel8-x86_64_v3
[+]              ^berkeley-db@18.1.40%gcc@12.1.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rhel8-x86_64_v3
[+]              ^gdbm@1.23%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]                  ^readline@8.2%gcc@12.1.1 build_system=autotools patches=bbf97f1 arch=linux-rhel8-x86_64_v3
[+]      ^gmake@4.4.1%gcc@12.1.1~guile build_system=autotools arch=linux-rhel8-x86_64_v3
[+]      ^pkgconf@1.9.5%gcc@12.1.1 build_system=autotools arch=linux-rhel8-x86_64_v3
[+]      ^zstd@1.5.5%gcc@12.1.1+programs build_system=makefile compression=none libs=shared,static arch=linux-rhel8-x86_64_v3
```

which shows I've installed the package dependencies already.

And the resulting concretized `sperr` from `spack spec sperr` is just `sperr@0.7.1%gcc@12.1.1~ipo+openmp+shared+utilities build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64_v3` (and no `zstd`).